### ISSUE TITLE
[BUG] Files matched with `.eslintignore` appeared in the list of violations with severity 1

### ISSUE THREAD

**Issue by constantine-prokopenko-vrp** [forcedotcom/sfdx-scanner#1101]
**Describe the bug**
Files listed in the `.eslintignore` appeared in the list of violations with high (1) severity. Below you can find and example
```
{
        "engine": "eslint-custom",
        "filename": "force-app/main/default/lwc/jsconfig.json",
        "violations": [
            {
                "severity": 1,
                "message": "File ignored because of a matching ignore pattern. Use \"--no-ignore\" to override.",
                "ruleName": null,
                "category": "problem",
                "url": ""
            }
        ]
    },
```

**To Reproduce**

1. Create sfdx project
2. create .eslintrc file with the next body
```
{
    "extends": [
        "eslint:recommended",
        "@salesforce/eslint-config-lwc/recommended",
        "@locker/eslint-config-locker/recommended"
    ],
    "globals": {
        "$A": "readonly",
        "$": "readonly"
    }
}
```
3. Create .eslintignore with the following body
```
**/lwc/**/*.css
**/lwc/**/*.html
**/lwc/**/*.json
**/lwc/**/*.svg
**/lwc/**/*.xml
**/aura/**/*.auradoc
**/aura/**/*.cmp
**/aura/**/*.css
**/aura/**/*.design
**/aura/**/*.evt
**/aura/**/*.json
**/aura/**/*.svg
**/aura/**/*.tokens
**/aura/**/*.xml
**/aura/**/*.app
```
4. Create LWC component
5. Run `sfdx scanner:run -f json -o scanResult.json -t force-app --eslintconfig .eslintrc`
6. Review scanResult.json file

**Expected behavior**
Ignored files should not appear on the list of violations.

**Screenshots**
<img width="1628" alt="Screenshot 2023-06-29 at 15 17 59" src="https://github.com/forcedotcom/sfdx-scanner/assets/48758429/1ae4552f-fbdf-4488-afd0-538764bade98">

**Desktop (please complete the following information):**
 - OS: MacOS Ventura 13.4.1 (22F82)
 - Scanner Version: @salesforce/sfdx-scanner 3.13.0

**Comment by jfeingold35** [comment]
@constantine-prokopenko-vrp , thanks for logging this one. I'll try to reproduce it and get back to you as soon as I'm able.

**Comment by jfeingold35** [comment]
@constantine-prokopenko-vrp , I've reproduced it, as well as identified the cause.
Basically, what's happening is that we decompose your targeted directory into files and provide those to ESLint. ESLint will silently skip ignored files in a targeted directory, but throws the warning for ignored files that were directly targeted.
I'll get a bug logged for you.

**Comment by DougMidgley** [comment]
Issue relates to https://github.com/forcedotcom/sfdx-scanner/issues/810 also. 
While this error keep showing up, its impossible to get a clear list of real issues.

**Comment by jfeingold35** [comment]
@constantine-prokopenko-vrp , @DougMidgley , we've implemented a fix for this issue that will be released in the upcoming 3.15.0 release scheduled for next week.
I'm going to close this issue for now, and it can be re-opened if need be.

**Comment by constantine-prokopenko-vrp** [comment]
Sure, 
Many thanks for the response @jfeingold35.
